### DISCUSSION TITLE
dmaker.derh.22l

### DISCUSSION THREAD

**Discussion by sll1001** [merdok/homebridge-miot#479] [Q&A]
I have successfully configure homebridge-miot v1.6.0 to control my dehumidifier. One setup that took me sometime to figure out is to setup the Tank Full alarm. From the documentation I figured one should use property monitor to get the Tank Full alarm and present it as a homekit occupancy event. However, setting it up directly using the configure UI does not work. After some trail and error, I figured that editing the configuration to the following works:

{
                    "name": "Living Room Dehumidifier",
                    "ip": "172.16.88.xx",
                    "token": "xxxxxx",
                    "deviceId": "xxxx",
                    "model": "dmaker.derh.22l",
                    "pollingInterval": 10,
                    "deepDebugLog": true,
                    "onlyMainService": true,
                    "buzzerControl": false,
                    "ledControl": false,
                    "childLockControl": false,
                    "modeControl": true,
                    "suppressAutoServiceCreation": [
                        "temperature",
                        "relativeHumidity",
                        "illumination"
                    ],
                    "propertyMonitor": [
                        {
                            "property": "dehumidifier:fault",
                            "value": 1,
                            "name": "LRD Tank Full"
                        }
                    ],
                    "fanLevelControl": false
}

In particular, the propertyMonitor param must be in this order: property, value, name. The UI writes the param in this order: property, name, value, does not work. May be it is a bug in parsing the configuration data.

The other thing I noticed is that when the tank is full, the Tank Full occupancy sensor on Homekit will oscillate between True and False every 30 seconds. Is it possible to keep it constantly in True state until the tank full alarm is clear?

Thanks for the great work. Previously I have modified another xiaomi plugin to work with this dehumidifier, now I can use this plugin without hacking the code!

**Comment by merdok** [comment]
Well, that with the config is unfortunately an issue with the ui and i already tried to do my best to fix that, but it is quite hard... Hence i usually recommended to manually update the json as this seems more safe!

It might be that the issue with the occupancy sensor is that it requires to be a specific value, in version 1.6.1 you will be able to set the equal or below as a modifier for the property monitor hence that might help with your issue. Please let me know after the update if it got better.

**Reply by sll1001** [reply]
Thank you for your suggestions.
On the issue with the occupancy sensor, I got a bit further. The reason it is oscillating between True and False is because there is more than one fault code at the time, namely the same property gives out 1 (tank full) and 5 (clean filter) and it will return either value on each read! I don't know what is the best way to deal with this. I guess implementing the event interface might help.

I found out what was happening by reading the property using the CLI from python-miio. I tried using your built-in miot command but could not figure out the format to send a raw command. Would be grateful if you can provide an example of sending raw command using the tank full as an example!!

**Reply by merdok** [reply]
You can have a look at the robot cleaner room cleaning guide. There are a couple of examples there.
